src/Experimental/isomorphism.jl

```diff
@@ -7,6 +7,21 @@ const IsomorphismProblem = IsomorphismProblemType()
 const SubGraphIsomorphismProblem = SubGraphIsomorphismProblemType()
 const InducedSubGraphIsomorphismProblem = InducedSubGraphIsomorphismProblemType()
 
+"""
+    IsomorphismAlgorithm
+
+An abstract type used for method dispatch on isomorphism functions.
+"""
+abstract type IsomorphismAlgorithm end
+
+"""
+    VF2
+
+An empty concrete type used to dispatch to [`vf2`](@ref) isomorphism functions.
+"""
+struct VF2 <: IsomorphismAlgorithm end
```

Comment:
Instead of having a type `VF2`, you could create a type `VF2Type` (or similar name) and then then declare a constant `const VF2 = VF2Type()`.  Then a user does not have do write the brackets in `VF2()`. It is not a big difference, but this pattern is for example also used in [base/sort.jl](https://github.com/JuliaLang/julia/blob/ebfa1d5b286a0415882a988282bc965657e3670b/base/sort.jl) for specifying the sorting algorithm and I used the same pattern when declaring `IsomorphismProblem`, `SubGraphIsomorphismProblem` and so on.

Reply:
We should be consistent with what we're already doing in LightGraphs, which is to create the instances using `()`. If we want to change it across the board within LG, that's probably a separate discussion.